.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,25 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.46.2
+          args: --timeout 5m --verbose
```

Comment:
Instead of using args, we could create a `.golangci.yaml` file to store the configuration and then it would be reused when running it both locally and on the CI. This way we would not have to keep it in sync.

https://golangci-lint.run/usage/configuration/#config-file